src/main/src/main/java/org/geoserver/security/SecureCatalogImpl.java

```diff
@@ -1524,6 +1524,24 @@ public void setDefaultDataStore(WorkspaceInfo workspace, DataStoreInfo defaultSt
                 filteredWrapped, com.google.common.base.Predicates.<T>notNull());
     }
 
+    @Override
+    public <T extends CatalogInfo> CloseableIterator<T> list(
+            Class<T> of,
+            Filter filter,
+            Integer offset,
+            Integer count,
+            SortBy sortBy1,
```

Comment:
This is clumsy, the time we need a three way sort for whatever reason are we going to have a new public method for it? The underlying API allows for a open ended list of sort values to be used, it's just easier to change the existing list method to take a "SortBy..." instead.
If you're out of time on this one I'll be happy to take a stab at it (instead of losing the PR again like last time).

Reply:
I'm still doing the changes that you requested, it was not really ready for review, but useful anyway, thanks. I'll tell you when is ready again for your review, to not waste your time. 
I've done the replacement with variable arguments `SortBy...` but there is still some part not ready for the behaviour of null (and the resulting SortBy[] {null} ), I need to investigate better. I guess this was the reason there was `SortBy` instead of `SortBy...`